lib/aws.js

```diff
@@ -33,8 +33,7 @@ AWS.util.update(AWS.Config.prototype.keys, {
     var credentials = null;
     new AWS.CredentialProviderChain([
       function () { return new AWS.EnvironmentCredentials('AWS'); },
-      function () { return new AWS.EnvironmentCredentials('AMAZON'); },
-      function () { return new AWS.SharedIniFileCredentials(); }
```

Comment:
I don't think there's a way to implement option 3 in a way that solves the issue without requiring users to change their flow.

If we added a configuration option to the provider to disable assume role, that could work. Something like `disableAssumeRole` that would still throw an error if the `roleArn` was found for the profile. I'd suggest defaulting it to false and setting it to true when it is called here.

The argument could be made for another credential provider to handle this, but I also think that having the `SharedIniFileCredentials` object handle these credentials from the ini file makes sense.

The impact is pretty small, only potentially causing an extra `STS` request when the SDK is required for the first time.